spring-kafka/src/test/java/org/springframework/kafka/support/micrometer/ObservationTests.java

```diff
@@ -449,6 +456,65 @@ void kafkaAdminNotRecreatedIfBootstrapServersSameInProducerAndAdminConfig(
 		assertThat(template.getKafkaAdmin()).isSameAs(kafkaAdmin);
 	}
 
+	// https://github.com/spring-cloud/spring-cloud-stream/issues/3095#issuecomment-2707075861
+	// https://github.com/spring-projects/spring-kafka/issues/3786
+	@Test
+	void verifyKafkaRecordSenderContextTraceParentHandling() {
+		String initialTraceParent = "traceparent-from-previous";
+		String updatedTraceParent = "traceparent-current";
+		ProducerRecord<Integer, String> record = new ProducerRecord<>("test-topic", "test-value");
+		record.headers().add("traceparent", initialTraceParent.getBytes(StandardCharsets.UTF_8));
+
+		// Create the context and update the traceparent
+		KafkaRecordSenderContext context = new KafkaRecordSenderContext(
+				record,
+				"test-bean",
+				() -> "test-cluster"
+		);
+		context.getSetter().set(record, "traceparent", updatedTraceParent);
+
+		Iterable<Header> traceparentHeaders = record.headers().headers("traceparent");
+		List<String> headerValues = StreamSupport.stream(traceparentHeaders.spliterator(), false)
+				.map(header -> new String(header.value(), StandardCharsets.UTF_8))
+				.collect(Collectors.toList());
+
+		// Verify there's only one traceparent header and it contains the updated value
+		assertThat(headerValues).containsExactly(updatedTraceParent);
+	}
+
+	// https://github.com/spring-cloud/spring-cloud-stream/issues/3095#issuecomment-2707075861
+	// https://github.com/spring-projects/spring-kafka/issues/3786
```

Comment:
I don't think there is a reason in this comments.
We always can find out that from the commit history.